common/djangoapps/third_party_auth/tests/test_provider.py

```diff
@@ -84,6 +90,41 @@ def test_providers_displayed_for_login(self):
         self.assertNotIn(no_log_in_provider.provider_id, provider_ids)
         self.assertIn(normal_provider.provider_id, provider_ids)
 
+    def test_providers_enabled_for_site(self):
+        """
+        Test the logic of the enabled_for_current_site property to ensure that a provider can only be
+        used when making requests in the context of its corresponding site.
+        """
+        site_a, __ = Site.objects.get_or_create(domain=SITE_DOMAIN_A, name=SITE_DOMAIN_A)
+        site_b, __ = Site.objects.get_or_create(domain=SITE_DOMAIN_B, name=SITE_DOMAIN_B)
+
+        default_site_provider = self.configure_google_provider(visible=True, enabled=True)
+        default_site_disabled_provider = self.configure_google_provider(visible=True, enabled=False)
+        site_provider = self.configure_google_provider(visible=True, enabled=True, site=site_a)
+        site_disabled_provider = self.configure_google_provider(visible=True, enabled=False, site=site_a)
+
+        site_b_provider = self.configure_google_provider(visible=True, enabled=True, site=site_b)
+
+        def enabled_for_current_site(assertions):  # pylint: disable=missing-docstring
+            for auth_provider, value in assertions:
+                self.assertEqual(auth_provider.enabled_for_current_site, value)
+
+        enabled_for_current_site((
```

Comment:
@bdero looks like you're trying to evaluate a number of cases in single test. Usually we (both edX and Opencraft) tend to write such kind of tests as parameterized tests using `ddt`. The benefit of doing so is that while still sharing the code, there are one test for every case, so when some of the conditions are not met you get a better understanding what exactly went wrong (which pair caused an error) and all the other cases are tested too.

In this case, you should have something like this:

    @ddt.data(
        (True, None, True),
        (False, None, False),
        (True, site_a, True),
        (False, site_a, False),
        ....
    )
    @ddt.unpack
    test_providers_enabled_for_site(enabled, site, expected_enabled):
        provider = self.configure_google_provider(visible=True, enabled=enabled, site=site)
        self.assertEqual(provider.enabled_for_current_site, expected_enabled)

   # and the same one wrapped into with_site_configuration_context(SITE_DOMAIN_A)